clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -230,6 +231,11 @@ public SslEngineFactory sslEngineFactory() {
         }
     }
 
+    @Override
+    public void close() {
```

Comment:
It seems to me adding a KIP for this interface change would be an overkill. But, I'm ok to write a KIP if it worth it.

Reply:
I don't think we consider `SslFactory` a public API, so I think we are probably good here.